hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/keyvalue/TestTarContainerPacker.java

```diff
@@ -188,13 +184,139 @@ public void pack() throws IOException, CompressorException {
     assertExampleChunkFileIsGood(
         Paths.get(destinationContainerData.getChunksPath()));
     Assert.assertFalse(
-        "Descriptor file should not been exctarcted by the "
+        "Descriptor file should not have been extracted by the "
             + "unpackContainerData Call",
         destinationContainer.getContainerFile().exists());
     Assert.assertEquals(TEST_DESCRIPTOR_FILE_CONTENT, descriptor);
+  }
+
+  @Test
+  public void unpackContainerDataWithValidRelativeDbFilePath()
+      throws Exception {
+    //GIVEN
+    KeyValueContainerData sourceContainerData =
+        createContainer(SOURCE_CONTAINER_ROOT);
+
+    String fileName = "sub/dir/" + TEST_DB_FILE_NAME;
+    File file = writeDbFile(sourceContainerData, fileName);
+    String entryName = TarContainerPacker.DB_DIR_NAME + "/" + fileName;
+
+    File containerFile = packContainerWithSingleFile(file, entryName);
+
+    // WHEN
+    unpackContainerData(containerFile);
+
+    // THEN
+    assertExampleMetadataDbIsGood(file.toPath().getParent());
+  }
+
+  @Test
+  public void unpackContainerDataWithValidRelativeChunkFilePath()
+      throws Exception {
+    //GIVEN
+    KeyValueContainerData sourceContainerData =
+        createContainer(SOURCE_CONTAINER_ROOT);
+
+    String fileName = "sub/dir/" + TEST_CHUNK_FILE_NAME;
+    File file = writeChunkFile(sourceContainerData, fileName);
+    String entryName = TarContainerPacker.CHUNKS_DIR_NAME + "/" + fileName;
+
+    File containerFile = packContainerWithSingleFile(file, entryName);
+
+    // WHEN
+    unpackContainerData(containerFile);
+
+    // THEN
+    assertExampleChunkFileIsGood(file.toPath().getParent());
```

Comment:
The assertion here is done on source container itself, is this intentional?

Reply:
Good point.

I don't recall exactly why I added this assertion.  I can say that the main goal of the valid/invalid test cases is to verify that they are accepted/rejected by `unpackContainerData` (by way of not throwing or throwing exception).  The subsequent assertion is not really necessary, as the pack/unpack process is validated in `@Test pack()`.

Now that you pointed this out, I tweaked the test a bit to be able to verify the chunk/db file for these cases, too.  Also added post-test cleanup.

Thanks for the review.